sklearn/preprocessing/tests/test_data.py

```diff
@@ -2227,6 +2227,15 @@ def test_categorical_encoder_dtypes_pandas():
     assert_array_equal(enc.transform(X).toarray(), exp)
 
 
+def test_categorical_encoder_warning():
+    # check no warning is raised
+    with warnings.catch_warnings(record=True) as w:
+        enc = CategoricalEncoder()
+        X = [['Male', 1], ['Female', 3], ['Female', 2]]
```

Comment:
Please move the above two lines outside of the `with` statement,
```py
X = ...
enc = ..
with ...
```
The class initialization currently doesn't do anything that could raise a warning and this would make more explicit the tested part.

Also, the third element in the `X` is not necessary for this example as far as I understand and could be removed..

Reply:
Updated, please take a look.